### PR TITLE
✨ Validate win

### PR DESCRIPTION
**Pull request by LouisaStrunk02**
Changes:

- checks for win
- asks for playing again

### CODE REVIEW

File: MainGame.py [hunk 1]
```diff
@@ -49,13 +52,33 @@ def get_column_input(self, player_name: str) -> int:
             except ValueError:
                 print("Please enter an integer.")
 
-    def round(self) -> None:
+    def round(self) -> None or Player:
+        winner: Player
         self.turn_number += 1
+
+        if self.turn_number > 42:
```
Comment (DevBastiS) [inline MainGame.py:59]: 42 is a "magic number". Please move it to a constant and give it a speaking name.

**Review by DevBastiS — APPROVED** [review]

File: MainGame.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
+from asyncio import constants
```
Comment (stefanluth) [inline MainGame.py:1]: This import is wrong :) asyncio is for handling asynchronous tasks.

File: MainGame.py [hunk 3]
```diff
@@ -49,13 +53,34 @@ def get_column_input(self, player_name: str) -> int:
             except ValueError:
                 print("Please enter an integer.")
 
-    def round(self) -> None:
+    def round(self) -> None or Player:
```
Comment (stefanluth) [inline MainGame.py:56]: `None or Player` is not a valid type :) Use [Optional](https://docs.python.org/3/library/typing.html#typing.Optional) or [Unions](https://docs.python.org/3/library/typing.html#typing.Union) instead

File: StatusValidator.py [hunk 4]
```diff
@@ -1,3 +1,67 @@
+from Board import Board
+from Player import Player
+
 class StatusValidator:
     def __inint__(self):
         pass
+
+    def is_win(board: Board, player: Player) -> bool:
```
Comment (stefanluth) [inline StatusValidator.py:8]: The first parameter should be `self`

**Review by stefanluth — CHANGES_REQUESTED** [review]

**Review by MKrips — APPROVED** [review]

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: StatusValidator.py [hunk 5]
```diff
@@ -1,3 +1,67 @@
...
+
 class StatusValidator:
-    def __inint__(self):
+    def __init__(self):
         pass
+
+    def is_win(self, board: Board, player: Player) -> bool:
+        is_horizontal_win = StatusValidator.horizontal_win(board,player)
+        is_vertical_win = StatusValidator.vertical_win(board, player)
+        is_diagonal_win = StatusValidator.diagonal_win(board, player)
```
Comment (stefanluth) [inline StatusValidator.py:11]:
```suggestion
        is_horizontal_win = self.horizontal_win(board,player)
        is_vertical_win = self.vertical_win(board, player)
        is_diagonal_win = self.diagonal_win(board, player)
```

Comment (stefanluth) [reply StatusValidator.py:11]: Other occurrences as well

File: MainGame.py [hunk 6]
```diff
@@ -32,8 +33,10 @@ def turn(self, player: Player) -> None:
 
             if is_valid_turn:
                 self.board.add_coin_to_board(column, player.symbol)
-
-                return
+                player_is_winner = StatusValidator.is_win(self.board, player)
```
Comment (stefanluth) [inline MainGame.py:36]: This won't work because `StatusValidator` is not an instance of `StatusValidator`, which means the `self` parameter is missing :)

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: MainGame.py [hunk 7]
```diff
@@ -49,13 +52,34 @@ def get_column_input(self, player_name: str) -> int:
             except ValueError:
                 print("Please enter an integer.")
 
-    def round(self) -> None:
+    def round(self) -> None | Player:
+        ANZAHL_FELDER = 42
```
Comment (stefanluth) [inline MainGame.py:56]: Place constants at the beginning of the file.

File: MainGame.py (hunk 6 above)
Comment (LouisaStrunk02) [reply MainGame.py:36]: Do I need this when using static methods?

Comment (stefanluth) [reply MainGame.py:38]: No :D

**Pull request merged**